internal/config/toml.go

```diff
@@ -31,8 +31,11 @@ type statusSection struct {
 
 // etcd3Section holds the optional `etcd3` fragment
 type etcd3Section struct {
-	Endpoints    []string `toml:"endpoints"`
-	TxnTimeoutMs *uint64  `toml:"transaction_timeout_ms"`
+	Endpoints         []string `toml:"endpoints"`
+	TxnTimeoutMs      *uint64  `toml:"transaction_timeout_ms"`
+	ClientCertPubPath string   `toml:"client_cert_pub_path"`
+	ClientCertKeyPath string   `toml:"client_cert_key_path"`
+	CAsTrustedPath    string   `toml:"cas_trusted_path"`
```

Comment:
I think this configuration option does not belong here, it is my understanding that this shouldn't be required for using TLS client certificates.

I guess you are having some troubles with an untrusted server certificate. If so, I suggest adjusting your deployment instead so that the relevant CA certificate ends up in the proper CA bundle/location instead.

Reply:
In the case of client authentication against the DCE by means of a certificate, this configuration is necessary.
The problem of the trusted CA's can be solved by updating the container in the usual /etc/ssl/certs path, but I think that leaving the location of the list of trusted CA's configurable in this way gives the project greater configuration flexibility.
If you still don't see it as adequate, I'll still modify it.
Client authentication against the tDCS by certificate is documented here: https://etcd.io/docs/v3.2/op-guide/security/#basic-setup

Reply:
> In the case of client authentication against the ETCD by means of a certificate, this configuration is necessary.

I'm aware of the `etcd` CLI flags on that docpage, but could you please detail what you have found this to be necessary for in your scenario instead?

It is my understanding that, if you already have a properly setup CA bundle in your environment and the server certificate properly signed through a chain of trust to a trusted CA, then you won't need this.

In particular, this option should concern the trust-check that the **client** performs on the server certificate, which is unrelated to the authentication-check that the **server** performs on the client certificate (to the best of my understanding).

Reply:
Goodnight !!

Currently my environment has a secure ETCD cluster configured to require client authentication via certificate (https://etcd.io/docs/v3.2/op-guide/security/#basic-setup). It is for this reason, and under these circumstances, that the client is required to present a valid certificate for the cluster that identifies it (to the client). The ETCD cluster will receive that certificate in the authentication phase, and therefore the machines that run airlock to manage their reboots, when they communicate with the ETCD cluster to establish the locks, they will have to present their credentials, and that is why it must be possible to configure these certificates in the containers that the client machines of the cluster execute.

I don't know if I have explained myself, but if you need any other clarification, count on me.

Thank you very much.

Greetings!!

Reply:
Good morning!!

Could you read the justification you asked me for?

Thanks in advance.

Greeting!!

Reply:
Yes, thanks for the reply.

To the best of my knowledge I do overall understand client-certificate authentication and the setup you are trying to achieve, no problem with that. What I'm focusing on is that `CAsTrustedPath` does not seem to belong to the client-certificate logic, and thus it should not be added here.
I can just guess why you are trying to add **this specific option** (not the others, which are indeed part of the client-certificate requirements and are fine in this PR), but so far you didn't really clear that out.

So let me ask this explicitly: can you remove this `CAsTrustedPath` field from here (and the other places where it is referenced) and make your target work with just the rest of the logic currently existing in this PR?

Reply:
Done!!

Reply:
Hello @lucab!!

I have made the change you requested. What do you think now?

Greeting.